File: leetcode/java/ReverseWordsinaString.java
/**
 * Created by Robbert on 1/30/15.
 * Reverse Words in a String
 * https://oj.leetcode.com/problems/reverse-words-in-a-string/
 */
public class ReverseWordsinaString {
    public String reverseWords(String s) {
        if (s == null || s.length() == 0) {
            return s;
        }

        String[] content = s.split(" ");
        StringBuilder sb = new StringBuilder();
        for (int i = content.length - 1; i >= 0; i--) {
            if (!content[i].equals("")) {
                sb.append(content[i]).append(" ");
            }
        }

        return sb.length() == 0 ? "" : sb.substring(0, sb.length() - 1);
    }
}

File: leetcode/python/reverse_words_in_a_string.py
class Solution:
    # @param s, a string
    # @return a string
    def reverseWords(self, s):
        if not s:
            return ""
        else:
            return ' '.join(s.strip().split()[::-1])
    
    def reverseWords(self, s):
        if not s:
            return ""
        s = s.strip().split()

        lo, hi = 0, len(s)-1
        while lo <= hi:
            s[lo], s[hi] = s[hi], s[lo]
            lo, hi = lo+1, hi-1

        return ' '.join(s)